server/README.md

```diff
@@ -93,7 +94,101 @@ curl -X POST -H 'Accept: application/sparql-results+json' -H 'Content-Type: appl
 curl -X POST -H 'Content-Type: application/sparql-update' --data 'DELETE WHERE { <http://example.com/s> ?p ?o }' http://localhost:7878/update
 ```
 
-You could easily build your own Docker image by running `docker build -t oxigraph server -f server/Dockerfile .` from the root directory.
+### Run the Web server with basic authentication
+
+It can be useful to make Oxigraph SPARQL endpoint available publicly, with a layer of authentication on `/update` to be able to add data.
+
+You can do so by using a nginx basic authentication in an additional docker container with `docker-compose`. First create a `nginx.conf` file:
+
+```nginx
+daemon off;
+events {
+    worker_connections  1024;
+}
+http {
+    server {
+        server_name localhost;
+        listen 7878;
+        rewrite ^/(.*) /$1 break;
+        proxy_ignore_client_abort on;
+        proxy_set_header  X-Real-IP  $remote_addr;
+        proxy_set_header  X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header  Host $http_host;
+        proxy_set_header Access-Control-Allow-Origin "*";
+        location ~ ^(/|/query)$ {
+            proxy_pass http://oxigraph:7878;
+            proxy_pass_request_headers on;
+        }
+        location ~ ^(/update|/store)$ {
+            auth_basic "Oxigraph Administrator's Area";
+            auth_basic_user_file /etc/nginx/.htpasswd; 
+            proxy_pass http://oxigraph:7878;
+            proxy_pass_request_headers on;
+        }
+    }
+}
+```
+
+Then a `docker-compose.yml` in the same folder, you can change the default user and password in the `environment` section:
+
+```yaml
+version: "3"
+services:
+  oxigraph:
+    image: ghcr.io/oxigraph/oxigraph
+    ## To build from local source code:
+    # build:
+    #   context: .
+    #   dockerfile: server/Dockerfile
+    volumes:
+      - ./data:/data
+
+  nginx-auth:
+    image: nginx:1.21.4
+    environment:
+      - OXIGRAPH_USER=oxigraph
+      - OXIGRAPH_PASSWORD=oxigraphy
+    volumes:
+      - ./nginx.conf:/etc/nginx/nginx.conf
+      ## For multiple users: uncomment this line to mount a pre-generated .htpasswd 
+      # - ./.htpasswd:/etc/nginx/.htpasswd
+    ports:
+      - 7878:7878
+    entrypoint: "bash -c 'echo -n $OXIGRAPH_USER: >> /etc/nginx/.htpasswd && echo $OXIGRAPH_PASSWORD | openssl passwd -stdin -apr1 >> /etc/nginx/.htpasswd && /docker-entrypoint.sh nginx'"
+```
+
+Once the `docker-compose.yaml` and `nginx.conf` are ready, start the Oxigraph server and nginx proxy for authentication on http://localhost:7878:
+
+```sh
+docker-compose up
+```
+
+To make an update to the graph, first change `$OXIGRAPH_USER` and `$OXIGRAPH_PASSWORD`, or set the environment variables, then run:
```

Comment:
Maybe make clear that the `curl` command is an example to show how auth work, not a mandatory step to make auth work?